keanu-project/src/test/java/io/improbable/keanu/util/dot/WriteDotTest.java

```diff
@@ -0,0 +1,166 @@
+import java.util.List;
+
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertTrue;
+
+public class WriteDotTest {
+
+    private static BayesianNetwork gaussianNet;
+    private static BayesianNetwork booleanNet;
+    private static String testFileDir = "dottestdir";
+    private static String gaussianDotOutput;
+    private static String boolDotOutput;
+    private static String vertexDegree1DotOutput;
+    private static String vertexDegree2DotOutput;
+    private static String boolVertexLabel = "/BoolResult";
+    private static GaussianVertex gaussianV;
+    private static DoubleVertex doubleV;
+
+    @BeforeClass
+    public static void setup() throws IOException {
+
+        String gaussianDotFileName = testFileDir + "/gaussianDot.dot";
+        String boolDotFileName = testFileDir + "/boolDot.dot";
+        String vertexDegree1DotFileName = testFileDir + "/VertexOutput1.dot";
+        String vertexDegree2DotFileName = testFileDir + "/VertexOutput2.dot";
+
+        double mu = 0;
+        double sigma = 1;
+        gaussianV = new GaussianVertex(mu, sigma);
+        doubleV = new ConstantDoubleVertex(5.43);
+        DoubleVertex result = gaussianV.multiply(doubleV);
+        gaussianNet = new BayesianNetwork(result.getConnectedGraph());
+        WriteDot.outputDot(gaussianDotFileName, gaussianNet);
+        gaussianDotOutput = new String(Files.readAllBytes(Paths.get(gaussianDotFileName)), Charset.defaultCharset());
+
+
+        ConstantBoolVertex boolV = new ConstantBoolVertex(true);
+        ConstantIntegerVertex intV = new ConstantIntegerVertex(5);
+        DoubleVertex doubleV2 = new ConstantDoubleVertex(5.43);
+        BoolVertex comparisonV = intV.greaterThan(doubleV2);
+        BoolVertex result2 = boolV.or(comparisonV);
+        result2.setLabel(boolVertexLabel);
+        booleanNet = new BayesianNetwork(result2.getConnectedGraph());
+        WriteDot.outputDot(boolDotFileName, booleanNet);
+        boolDotOutput = new String(Files.readAllBytes(Paths.get(boolDotFileName)), Charset.defaultCharset());
+
+        WriteDot.outputDot(vertexDegree1DotFileName, gaussianV, 1);
+        vertexDegree1DotOutput = new String(Files.readAllBytes(Paths.get(vertexDegree1DotFileName)), Charset.defaultCharset());
+        WriteDot.outputDot(vertexDegree2DotFileName, gaussianV, 2);
+        vertexDegree2DotOutput = new String(Files.readAllBytes(Paths.get(vertexDegree2DotFileName)), Charset.defaultCharset());
+    }
+
+    @Test
+    public void filesGetWrittenAndAreNotOverwritten() throws IOException {
+        String fileName = "dottestfolder/test_file.dot";
+        File outputFile = new File(fileName);
+        FileUtils.deleteDirectory(outputFile.getParentFile());
+
+        WriteDot.outputDot(fileName, gaussianNet);
+        assertTrue(outputFile.exists());
+
+        WriteDot.outputDot(fileName, booleanNet);
+        assertTrue(outputFile.getParentFile().listFiles().length == 2);
```

Comment:
you generally get more expressive error messages if you use hamcrest matchers, e.g.

        assertThat(outputFile.getParentFile().listFiles(), arrayWithSize(2));

Reply:
Ok, will use that.